src/main/java/it/gov/pagopa/payhub/activities/activity/ingestionflow/debtposition/SynchronizeIngestedDebtPositionActivityImpl.java

```diff
@@ -0,0 +1,40 @@
+package it.gov.pagopa.payhub.activities.activity.ingestionflow.debtposition;
+
+import it.gov.pagopa.payhub.activities.service.debtposition.DebtPositionSearchService;
+import it.gov.pagopa.pu.debtposition.dto.generated.DebtPositionDTO;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Slf4j
+@Lazy
+@Service
+public class SynchronizeIngestedDebtPositionActivityImpl implements SynchronizeIngestedDebtPositionActivity {
+
+    private final DebtPositionSearchService debtPositionSearchService;
+
+    public SynchronizeIngestedDebtPositionActivityImpl(DebtPositionSearchService debtPositionSearchService) {
+        this.debtPositionSearchService = debtPositionSearchService;
+    }
+
+    @Override
+    public String synchronizeIngestedDebtPosition(Long ingestionFlowFileId) {
+        log.info("Synchronizing all debt positions related to ingestion flow file id {}", ingestionFlowFileId);
+        List<DebtPositionDTO> debtPositions = debtPositionSearchService.getAllDebtPositionsByIngestionFlowFileId(ingestionFlowFileId);
```

Comment:
don't accumulate all DebtPosition, but work them page-by-page.
on each page, you should invoke the API and then wait for all WF if there is a workflowId

ps: pageSize could still be a configuration read by the service instead, as the sort configuration